distributed/tests/test_core.py

```diff
@@ -886,6 +886,30 @@ async def remove_address():
     assert connect_finished.cancelled()
 
 
+@gen_test()
+async def test_remove_cancels_connect_before_task_running():
+    loop = asyncio.get_running_loop()
+    connect_finished = loop.create_future()
+
+    async def connect(*args, **kwargs):
+        await connect_finished
```

Comment:
```suggestion
    async def connect(*args, **kwargs):
        await asyncio.Future()  # Block until cancelled
```

Reply:
loop.create_future() is preferred to asyncio.Future()